openlibrary/plugins/openlibrary/lists.py

```diff
@@ -641,3 +644,28 @@ def get_active_lists_in_random(limit=20, preload=True):
     lists = f(limit=limit, preload=preload)
     # convert rawdata into models.
     return [web.ctx.site.new(xlist['key'], xlist) for xlist in lists]
+
+def create_list_preview(lst_key):
+    """Fetches the five first seeds of a list"""
+    lst = web.ctx.site.get(lst_key)
+    five_seeds = lst.seeds[0:5]
+    return five_seeds
+
+def create_preview_recommendation_text(lst_key):
+    lst = web.ctx.site.get(lst_key)
+    list_title = lst.name
+    list_owner = lst.get_owner()
+    recom_text = list_owner.displayname + "'s " + list_title
+    return recom_text
+
+class lists_preview(delegate.page):
+    path = r"(/people/[^/]+/lists/OL\d+L)/preview"
+
+    def GET(self, path):
+        image_bytes = overlay_covers_over_background(path)
+        binary_file = tempfile.NamedTemporaryFile()
+        binary_file.write(image_bytes)
+
+        web.header("Content-Type", "image/png")
+        with open(binary_file.name, 'rb') as f:
+            return delegate.RawText(f.read())
```

Comment:
Once this is all working, please try to substitute an in-memory binary file:
```suggestion
        web.header("Content-Type", "image/png")
        with io.BytesIO(overlay_covers_over_background(path)) as image_bytes:
            return delegate.RawText(image_bytes.getvalue())
```
If that works, then it should be faster than a disk-based file.  If it does not, then please stick with what works.